sources/StdlibUpgradeScripts.move

```diff
@@ -2,120 +2,139 @@ address StarcoinFramework {
-        }
+    /// deprecated, use `do_upgrade_from_v6_to_v7_with_language_version`.
+    public fun do_upgrade_from_v6_to_v7(sender: &signer) {
+        do_upgrade_from_v6_to_v7_with_language_version(sender, 2);
+    }
 
-        public entry fun upgrade_from_v7_to_v8(sender: signer) {
-            do_upgrade_from_v7_to_v8(&sender);
-        }
-        public fun do_upgrade_from_v7_to_v8(sender: &signer) {
-            {
-                let cap = Oracle::extract_signer_cap(sender);
-                GenesisSignerCapability::initialize(sender, cap);
-            };
-
-            {
-                let cap = NFT::extract_signer_cap(sender);
-                Account::destroy_signer_cap(cap);
-            };
-        }
+    public fun do_upgrade_from_v6_to_v7_with_language_version(sender: &signer, language_version: u64) {
+        // initialize the language version config.
+        Config::publish_new_config(sender, LanguageVersion::new(language_version));
+        // use STC Dao to upgrade onchain's move-language-version configuration.
+        OnChainConfigDao::plugin<STC, LanguageVersion::LanguageVersion>(sender);
+        // upgrade genesis NFT
+        GenesisNFT::upgrade_to_nft_type_info_v2(sender);
+    }
 
-        public entry fun upgrade_from_v11_to_v12(sender: signer) {
-            do_upgrade_from_v11_to_v12(&sender);
-        }
-        public fun do_upgrade_from_v11_to_v12(sender: &signer) {
-            {
-                GasSchedule::initialize(sender,GasSchedule::new_gas_schedule());
-                let address = @0x8c109349c6bd91411d6bc962e080c4a3;
-                EasyGas::initialize(sender,
-                    address,
-                    b"STAR",b"STAR",
-                    address);
-                Block::checkpoints_init(sender);
-            };
+    public entry fun upgrade_from_v7_to_v8(sender: signer) {
+        do_upgrade_from_v7_to_v8(&sender);
+    }
+    public fun do_upgrade_from_v7_to_v8(sender: &signer) {
+        {
+            let cap = Oracle::extract_signer_cap(sender);
+            GenesisSignerCapability::initialize(sender, cap);
+        };
+
+        {
+            let cap = NFT::extract_signer_cap(sender);
+            Account::destroy_signer_cap(cap);
+        };
+    }
+
+    public entry fun upgrade_from_v11_to_v12(sender: signer) {
+        do_upgrade_from_v11_to_v12(&sender);
+    }
+    public fun do_upgrade_from_v11_to_v12(sender: &signer) {
+        {
+            GasSchedule::initialize(sender,GasSchedule::new_gas_schedule());
+            let address = @0x8c109349c6bd91411d6bc962e080c4a3;
+            EasyGas::initialize(sender,
+                address,
+                b"STAR",b"STAR",
+                address);
+            Block::checkpoints_init(sender);
+        };
+    }
+
+    public entry fun upgrade_from_v12_to_v13(sender: signer) {
+        do_upgrade_from_v12_to_v13(&sender);
+    }
+    public fun do_upgrade_from_v12_to_v13(sender: &signer) {
```

Comment:
We could do a proper initialization when upgrading stdlib, like pushing_new_config then set the address list. Then we could burn the balance in a block triggered by  executing a DAO proposal.